src/zarr.cpp

```diff
@@ -587,6 +599,24 @@ zarr::Zarr::set_chunking(const ChunkingProps& props, const ChunkingMeta& meta)
     CHECK(planes_per_chunk_ > 0);
 }
 
+void
+zarr::Zarr::reset_state_()
+{
+    // reset state variables
+    writers_.clear();
+
+    // should be empty, but just in case
+    for (auto& [_, frame] : scaled_frames_) {
+        if (frame.has_value() && frame.value()) {
+            free(frame.value());
```

Comment:
The comment above provides some context, but this makes me ask the question who owns these video frames. I guess it's acquire's source? But `free` is called elsewhere in this implementation, so this isn't new, and thus is not a blocker for this PR.

Reply:
The Zarr object owns them. They're created in `scale_image()` (on line 260 to be exact. Even the full-res images are copies of the frames from the source.

Reply:
Oh thanks for clarifying. Resolved.